numpy/lib/npyio.py

```diff
@@ -728,20 +728,20 @@ def _savez(file, args, kwds, compress, allow_pickle=True, pickle_kwargs=None):
     zipf.close()
 
 
-def _getconv(dtype):
-    """ Find the correct dtype converter. Adapted from matplotlib """
+def _floatconv(x):
+    try:
+        return float(x)  # The fastest path.
+    except ValueError:
+        if '0x' in x:  # Don't accidentally convert "a" ("0xa") to 10.
+            try:
+                return float.fromhex(x)
+            except ValueError:
+                pass
+        raise  # Raise the original exception, which makes more sense.
 
-    def floatconv(x):
-        try:
-            return float(x)  # The fastest path.
-        except ValueError:
-            if '0x' in x:  # Don't accidentally convert "a" ("0xa") to 10.
-                try:
-                    return float.fromhex(x)
-                except ValueError:
-                    pass
-            raise  # Raise the original exception, which makes more sense.
 
+def _getconv(dtype):
+    """ Find the correct dtype converter. Adapted from matplotlib """
     typ = dtype.type
     if issubclass(typ, np.bool_):
         return lambda x: bool(int(x))
```

Comment:
I think you may as well extract this `lambda` and the other two in this this function as `_boolconv`, `_complexconv`, and `_intconv`; right now your optimization in this PR doesn't work for these types.

Reply:
done, and extracted all converters to a global table for good measure.